### ISSUE TITLE
Difference in nested resource wrapping between 0.1.x and 0.2.x

### ISSUE THREAD

**Issue by AlexVanderbist** [inertiajs/inertia-laravel#100]
I noticed a change in behaviour between version 0.1 and 0.2 that I'd like to address.

As of 0.2.0 responsables in the props will be recursively converted to arrays using `$responsable->toResponse()->getData()`. As discussed in #93, this supports Laravel's default behaviour for resources and nested resources will be converted to arrays correctly.

However, we want to avoid using `Resource::withoutWrapping()` because we _want_ the wrapping behaviour on collection resources. 

In version 0.1.x we solved this using the `$resource->resolve()` method. This caused the top-level resource to be converted to an array. Nested resources remained but Laravel made sure that they didn't receive the `data` wrapper when serialising.

In 0.2.0 this behaviour changed as resources are now recursively converted to arrays, without the context of being a nested resource. This causes a different result for the same data being passed to Laravel and Inertia.

Resource serialized by Laravel:
```php
        return new JsonResponse(UserResource::make(User::first())->resolve());
```
![image](https://user-images.githubusercontent.com/6287961/73168689-9ee48700-40fa-11ea-87a7-f4e43bee4329.png)

Resource serialized by Inertia:
```php
        return Inertia::render('users.Edit', [
            'user' => UserResource::make(User::first())->resolve(),
        ]);
```
![image](https://user-images.githubusercontent.com/6287961/73169245-f6cfbd80-40fb-11ea-83ef-62d77db7e398.png)

## Proposed solution

Don't convert `Responsable` `JsonResource`s to responses recursively? They can take care of themselves.

https://github.com/inertiajs/inertia-laravel/blob/fbd2afcadf96319a0403eb10f0a70b928ea009cf/src/Response.php#L64-L66
Could become something like this...
```php
if ($prop instanceof Responsable && ! $prop instanceof JsonResource) {
    $prop = $prop->toResponse($request)->getData();
}
```
Open for suggestions/discussion tho :)

**Comment by AlexVanderbist** [comment]
So I did some more digging today. Turns out that Laravel only wraps the resource when `toResponse` is called, not when serializing. So the proposed solution won't work either as that will serialize the resource and the wrapper won't be added when it's expected.

I'm gonna close this issue for now as it seems to originate from the way Laravel serializes resources—rather than how Inertia converts them to props.

Sorry for any confusion!
